contracts/governance/Vesting/DevelopmentVesting.sol

```diff
@@ -0,0 +1,216 @@
+    IERC20 public SOV;
+    ///@notice the owner of the vested tokens (timelock contract initially)
+    address public tokenOwner;
+
+    ///@notice the cliff. after this time period the tokens begin to unlock
+    uint public cliff;
+    ///@notice the duration. after this period all tokens will have been unlocked
+    uint public duration;
+    ///@notice the frequency. part of tokens will have been unlocked periodically
+    uint public frequency;
+
+    struct Schedule {
+        ///@notice the start date of the vesting
+        uint startDate;
+        ///@notice amount of vested tokens
+        uint amount;
+        ///@notice amount of already withdrawn tokens
+        uint withdrawnAmount;
+    }
+
+    Schedule[] public schedules;
+
+    ///@notice amount of tokens, these tokens can be withdrawn any time by an owner
+    uint public amount;
+
+    event TokensSent(address indexed caller, uint amount);
+    event TokensWithdrawn(address indexed caller, address receiver, uint amount);
+    event TokensVested(address indexed caller, uint amount);
+    event VestedTokensWithdrawn(address indexed caller, address receiver, uint amount);
+
+    /**
+     * @dev Throws if called by any account other than the token owner or the contract owner.
+     */
+    modifier onlyOwners() {
+        require(msg.sender == tokenOwner || isOwner(), "unauthorized");
+        _;
+    }
+
+    /**
+     * @notice setup the vesting schedule
+     * @param _SOV the SOV token address
+     * @param _tokenOwner the owner of the tokens
+     * @param _cliff the cliff in seconds
+     * @param _duration the total duration in seconds
+     * @param _frequency the withdrawal interval in seconds
+     */
+    constructor(address _SOV, address _tokenOwner, uint _cliff, uint _duration, uint _frequency) public {
+        require(_SOV != address(0), "SOV address invalid");
+        require(_tokenOwner != address(0), "token owner address invalid");
+
+        _setSchedule(_cliff, _duration, _frequency);
+
+        SOV = IERC20(_SOV);
+        tokenOwner = _tokenOwner;
+    }
+
+    /**
+     * @notice set token owner
+     * @param _tokenOwner the owner of the tokens
+     */
+    function setTokenOwner(address _tokenOwner) public onlyOwner {
+        require(_tokenOwner != address(0), "token owner address invalid");
+        tokenOwner = _tokenOwner;
+    }
+
+    /**
+     * @notice send tokens to this contract, these tokens can be withdrawn any time by an owner
+     * @param _amount the amount of tokens to send
+     */
+    function sendTokens(uint _amount) public {
+        require(_amount > 0, "amount needs to be bigger than 0");
+
+        //retrieve the SOV tokens
+        bool success = SOV.transferFrom(msg.sender, address(this), _amount);
+        require(success);
+
+        amount += _amount;
+
+        emit TokensSent(msg.sender, _amount);
+    }
+
+    //TODO onlyOwners or onlyOwner ?
+    /**
+     * @notice withdraws tokens
```

Comment:
maybe add the information that only tokens which were not subject to vesting can be withdrawn by this function

Reply:
done